### PR TITLE
cmake: link consumers of libclient with libcommon

### PR DESCRIPTION
**Pull request by tchaikov**

### CODE REVIEW

File: src/ceph_fuse.cc [hunk 1]
```diff
@@ -47,18 +47,18 @@ using namespace std;
...
-
-  assert(args.allocated);  // Checking fuse has realloc'd args so we can free newargv
-  free(argv);
+  assert(args.allocated);
+  fuse_opt_free_args(&args);
 }
+
 void usage()
 {
   cout <<
```
Comment (badone) [inline src/ceph_fuse.cc]: Should this be its own separate PR? Seems unrelated to the linking changes.

Comment (tchaikov) [inline src/ceph_fuse.cc]: the PR tries to fix the mem leak issues reported by valgrind, that's why i think this cleanup could be part of it.

Comment (badone) [inline src/ceph_fuse.cc]: OK. That's probably worth a mention in the commit message?

Comment (badone) [inline src/ceph_fuse.cc]: Doh! Of course the tracker mentions them. Sorry @tchaikov

**Comment by jcsp** [comment]
Valgrind issues look to be fixed by this: http://pulpito.ceph.com/jspray-2017-02-14_02:39:19-fs-wip-jcsp-testing-20170214-distro-basic-smithi/
